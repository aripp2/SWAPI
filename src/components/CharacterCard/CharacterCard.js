import React from 'react';
import './CharacterCard.scss';

const CharacterCard = ({ name, homeworld, population, species, films, updateFavs }) => {
  const character = { name, homeworld, population, species, films }
  const filmsList = films.map((film, i) => {
    return <li key={i}>{film}</li>
  })
  return (
      <div className="CharacterCard">
        <h3 className="CharacterCard_name">{name}</h3>
        <h4 className="CharacterCard_homeWorld">Homeworld: {homeworld}</h4>
        <h4 className="CharacterCard_population">Population: {population}</h4>
        <h4 className="CharacterCard_species">Species: {species}</h4>
        <h4 className="CharacterCard_films" >Related Films: </h4>
          <ul className="CharacterCard_films-ul">
            {filmsList}
          </ul>
        <button
          onClick={() => updateFavs(character)}
        >Favorite</button>
      </div>
    )
}

export default CharacterCard;

