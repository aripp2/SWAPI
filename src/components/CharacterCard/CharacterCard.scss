@import './src/variables.scss';

.CharacterCard {
  position: relative; 
  box-shadow: 0px 5px 5px 0px #e8e8e8, 0px 10px 10px 0px #e8e8e8;
  margin: 20px; 
  padding: 30px; 
  margin: 10px;
  min-height: 400px;
  min-width: 342px;
  max-width: 30%;

}

.CharacterCard_name { 
  position: absolute;
  top: 250px;
  left: 33px;
  font-size: 40px;
  font-weight: 900;
}

.CharacterCard_homeWorld-title {
  position: absolute;
  top: 91px;
  left: 35px;
  font-size: 10px;
  font-weight: 600; 
}

.CharacterCard_homeWorld {
  position: absolute;
  top: -8px;
  left: 62px;
  width: 150px;
  text-align: left;
  font-size: 20px;
  font-weight: 600;
}

.CharacterCard_species-title {
  position: absolute;
  top: 60px;
  left: 35px;
  font-size: 10px;
  font-weight: 600;
}

.CharacterCard_species {
  position: absolute;
  top: -8px;
  left: 46px;
  font-size: 20px;
  width: 142px;
  text-align: left;
  font-weight: 600;
}

.CharacterCard_population-title {
  position: absolute;
  top: 28px;
  left: 35px;
  font-size: 10px;
  font-weight: 600;
}
.CharacterCard_population {
  position: absolute;
  top: -8px;
  left: 60px;
  font-size: 20px;
  font-weight: 600;
}

.CharacterCard_character-btn:hover {
  background-color: $form-color;
}

.CharacterCard_character-btn {
  position: absolute;
  bottom: -26px;
  right: 19px;
  font-size: 20px;
  font-weight: 600;
  width: 239px;
  height: 50px;
  border-radius: 25px;
}

.CharacterCard_films {
  position: absolute;
  top: 130px;
  left: 35px;
  font-size: 10px;
  font-weight: 600; 
}
.CharacterCard_films-ul {
  position: absolute;
  top: 156px;
  left: 54px;
  font-size: 15px;
  font-size: 14px;
  font-weight: 600;
  list-style: none; 
}