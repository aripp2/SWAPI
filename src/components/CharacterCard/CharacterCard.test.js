import React from 'react';
import { shallow } from 'enzyme';
import CharacterCard from './CharacterCard'

describe('CharacterCard', () => {
  it('should match snapshot', () => {
    const wrapper = <CharacterCard /> 
    expect(wrapper).toMatchSnapshot(); 
  })
})