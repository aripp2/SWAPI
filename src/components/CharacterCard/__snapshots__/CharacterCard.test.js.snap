// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should match snapshot 1`] = `
<CharacterCard
  characters={
    Array [
      Object {
        "films": Array [
          "The Empire Strikes Back",
          "Attack of the Clones",
          "The Phantom Menace",
          "Revenge of the Sith",
          "Return of the Jedi",
          "A New Hope",
        ],
        "homeworld": "Tatooine",
        "name": "C-3PO",
        "population": "200000",
        "species": "Droid",
      },
      Object {
        "films": Array [
          "The Empire Strikes Back",
          "Attack of the Clones",
          "The Phantom Menace",
          "Revenge of the Sith",
          "Return of the Jedi",
          "A New Hope",
          "The Force Awakens",
        ],
        "homeworld": "Naboo",
        "name": "R2-D2",
        "pecies": "Droid",
        "population": "4500000000",
      },
    ]
  }
/>
`;
